src/char/int_guild.cpp

```diff
@@ -757,23 +741,64 @@ int inter_guild_CharOffline(uint32 char_id, int guild_id)
 	return 1;
 }
 
-// Initialize guild sql
-int inter_guild_sql_init(void)
-{
-	const char *filename[]={ DBPATH"exp_guild.txt", DBIMPORT"/exp_guild.txt"};
-	int i;
+const std::string GuildExpDatabase::getDefaultLocation() {
+	return std::string(db_path) + "/exp_guild.yml";
+}
+
+uint64 GuildExpDatabase::parseBodyNode(const YAML::Node &node) {
+	if (!this->nodesExist(node, { "Level", "Exp" })) {
+		return 0;
+	}
+
+	uint16 level;
+
+	if (!this->asUInt16(node, "Level", level))
+		return 0;
+
+	if (level == 0) {
+		this->invalidWarning(node, "The minimum guild level is 1.\n");
+		return 0;
+	}
+	if (level > MAX_GUILDLEVEL) {
+		this->invalidWarning(node["Level"], "Guild level %d exceeds maximum level %d, skipping.\n", level, MAX_GUILDLEVEL);
+		return 0;
+	}
+
+	uint64 exp;
+
+	if (!this->asUInt64(node, "Exp", exp))
+		return 0;
+
+	std::shared_ptr<s_guild_exp_db> guild_exp = this->find(level);
+	bool exists = guild_exp != nullptr;
+
+	if (!exists) {
+		guild_exp = std::make_shared<s_guild_exp_db>();
+		guild_exp->level = level;
+	}
+
+	guild_exp->exp = static_cast<t_exp>(exp);
+
+	if (!exists)
+		this->put(level, guild_exp);
+
+	return 1;
+}
+
+GuildExpDatabase guild_exp_db;
+
+// Initialize guild sql and read exp_guild.yml
+void inter_guild_sql_init(void) {
 	//Initialize the guild cache
 	guild_db_= idb_alloc(DB_OPT_RELEASE_DATA);
 	castle_db = idb_alloc(DB_OPT_RELEASE_DATA);
 
-	//Read exp file
-	for(i = 0; i<ARRAYLENGTH(filename); i++){
-		sv_readdb(db_path, filename[i], ',', 1, 1, MAX_GUILDLEVEL, exp_guild_parse_row, i > 0);
-	}
-
+	guild_exp_db.load();
+	if (guild_exp_db.size() < (MAX_GUILDLEVEL-1))
+		ShowWarning("Not enough entries for exp_guild, found %" PRIuPTR ", need at least %d.\n", guild_exp_db.size(), (MAX_GUILDLEVEL-1));	// 49 entries for pre-renewal, 50 for renewal
```

Comment:
Should be moved to loadingFinished.
Also check should be done by checking each level just to be save. :-)

Reply:
There are 49 entries in exp_guild.txt in pre-renewal folder (== guild max level is 50), 50 for renewal  (== guild max level is 51 regardless of MAX_GUILDLEVEL) but the value of MAX_GUILDLEVEL is the same for both
```
#define MAX_GUILDLEVEL 50 ///Max Guild level
```
I'm crazy or there is something wrong on the txt version.
Do you know the max guild level on KRO ?

Reply:
No idea about pre-renewal anymore, but for renewal this is 50 as far as I know.